src/js/BookReader/Toolbar/Toolbar.js

```diff
@@ -0,0 +1,383 @@
+
+    const $leftCol = $(`<div class="BRinfoLeftCol"></div>`);
+    if (br.thumbnail) {
+      $leftCol.append($(`
+        <div class="BRimageW">
+          <img src="${br.thumbnail}" alt="${escapeHTML(br.bookTitle)}" />
+        </div>`));
+    }
+
+    const $rightCol = $(`<div class="BRinfoRightCol">`);
+
+    // A loop to build fields
+    for (let {extraValueClass='', label, value} of br.metadata) {
+      $rightCol.append($(`
+        <div class="BRinfoValueWrapper">
+          <div class="BRinfoLabel">${label}</div>
+          <div class="BRinfoValue ${extraValueClass}">${value}</div>
+        </div>`));
+    }
+
+    const moreInfoText = br.bookUrlMoreInfo ? br.bookUrlMoreInfo : br.bookTitle;
+    if (moreInfoText && br.bookUrl) {
+      $rightCol.append($(`
+        <div class="BRinfoValueWrapper">
+          <div class="BRinfoMoreInfoWrapper">
+            <a class="BRinfoMoreInfo" href="${br.bookUrl}">
+              ${moreInfoText}
+            </a>
+          </div>
+        </div>`));
+    }
+
+    const $footer = $(`<div class="BRfloatFoot BRinfoFooter"></div>`);
+    const $children = $('<div class="BRinfoW mv20-lg">').append([
+      $leftCol,
+      $rightCol,
+      $('<br style="clear:both"/>'),
+    ]);
+
+    $infoDiv
+      .append($children, $footer)
+      .addClass('wide');
+  }
+
+  /**
+   * @return {number} (in pixels) of the toolbar height. 0 if no toolbar.
+   */
+  getToolBarHeight() {
+    const { $BRtoolbar } = this.br.refs;
+    if ($BRtoolbar && $BRtoolbar.css('display') === 'block') {
+      return ($BRtoolbar.outerHeight() + parseInt($BRtoolbar.css('top')));
+    } else {
+      return 0;
+    }
+  }
+}
+
+export function blankInfoDiv() {
+  // FIXME the <button> el is closed with a </a>
+  return $(`
+    <div class="BRfloat BRinfo">
+      <div class="BRfloatHead">About this book
+        <button class="floatShut" href="javascript:;" onclick="$.fn.colorbox.close();"><span class="shift">Close</span></a>
+      </div>
+      <div class="BRfloatBody">
+        <div class="BRfloatCover"></div>
+        <div class="BRfloatMeta">
+          <div class="BRfloatTitle">
+            <h2><a /></h2>
+          </div>
+        </div>
+      </div>
+      <div class="BRfloatFoot">
+        <a href="https://openlibrary.org/dev/docs/bookreader">About the BookReader</a>
+      </div>
+    </div>`);
+}
+
+export function blankShareDiv() {
+  // FIXME the <button> el is closed with a </a>
+  return $(`
+    <div class="BRfloat BRshare">
+      <div class="BRfloatHead">
+        Share
+        <button class="floatShut" href="javascript:;" onclick="$.fn.colorbox.close();"><span class="shift">Close</span></a>
+      </div>
+    </div>`);
+}
+
+/**
+ * Helper opens a popup window. On mobile it only opens a new tab. Used for share.
+ * @param {string} href
+ * @param {number} width
+ * @param {number} height
+ * @param {string} name
+ */
+export function createPopup(href, width, height, name) {
```

Comment:
do we expect this to be used outside of the file? is it possible to move this and `blankShareDiv` to its own place?

Reply:
I'd rather punt on this one; I think future us can decide if this is generally useful enough to place in a shared util file :+1: